dev-tools/reconfigurator-sp-updater/src/main.rs

```diff
@@ -377,6 +379,28 @@ enum Component {
         expected_active_version: ArtifactVersion,
         expected_inactive_version: ExpectedVersion,
     },
+    Rot {
+        /// whether we expect the "A" or "B" slot to be active
+        expected_active_slot: RotSlot,
```

Comment:
There's so many arguments here that I kind of thing we should make these named arguments instead of positional.  (I wouldn't object to making the SP ones named, too, if we feel that's better for consistency.)